### PR TITLE
Coward minigame in grey box lvl 

### PR DESCRIPTION
**Pull request by AlecatEnjoyer**
https://docs.google.com/document/d/1TK42grwHSTzWQ-xy8Wd4y12pLDItNDb2GrClAD9KXXI/edit#heading=h.9aksft4l3t6
 
scene: GreyBoxing

### CODE REVIEW

File: Assets/Scripts/Minigames/CowardMiniGame/GearBehavior.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        _doOnce = true;
+        _gearSizeNum = 1;
+        //_scrollable = false;
+        //GameObject _pcObject = GameObject.FindWithTag("Player");
+        //_pc = _pcObject.GetComponent<PlayerController>();
+        _rndr = this.GetComponent<Renderer>();
+    }
+
+    // Update is called once per frame
+    void FixedUpdate()
```
Comment (NGrinstead0101) [inline Assets/Scripts/Minigames/CowardMiniGame/GearBehavior.cs:43]: Consider adding more detailed comments to these update functions to describe what specifically they do

File: Assets/Scripts/Minigames/CowardMiniGame/GearCompletionCheck.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/*****************************************************************************
+// File Name :         GearCompletionCheck.cs
+// Author :            Mark Hanson
+// Creation Date :     5/27/2024
+//
+// Brief Description : Any function to do for the gears mini game will be found here. Includes swapping slots, Correct slot pattern with all bad ones, and selecting gears for each slot.
```
Comment (NGrinstead0101) [inline Assets/Scripts/Minigames/CowardMiniGame/GearCompletionCheck.cs:6]: Both this script and GearBehavior share this same description. Do they both serve similar functions?

File: Assets/Scripts/Minigames/CowardMiniGame/RipcordBehavior.cs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+{
+
+    [Header("UI Stuff")]
+    [SerializeField] private TextMeshPro _successfulPulls;
+
+    [Header("Points of Movement")]
+    [SerializeField] private float _maxReach;
+    [SerializeField]private GameObject _targetFollow;
+    [SerializeField] private float _speed;
+    public bool PressedE;
```
Comment (NGrinstead0101) [inline Assets/Scripts/Minigames/CowardMiniGame/RipcordBehavior.cs:26]: Does this need to be public or can it be a private serialized field?

File: Assets/Scripts/Minigames/CowardMiniGame/RipcordBehavior.cs [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        if(_pc.interact.IsPressed() && PressedE == true && _doOnce2 == true && _numReleased != 3)
+        {
+            Vector3 _previewerPoint = new Vector3(_relocatePoint.x, _relocatePoint.y, Random.Range(_maxReach,_relocatePoint.z));
+            Instantiate(_ReleasePreviewer, _previewerPoint, Quaternion.identity);
+            _doOnce2 = false;
+        }
+        if (PressedE == false && transform.position.z < _relocatePoint.z && transform.position.z != _relocatePoint.z || _pc.interact.IsPressed() && PressedE == true && transform.position.z < _maxReach && transform.position.z != _relocatePoint.z)
+        {
+            transform.position += new Vector3(0, 0, _speed * Time.deltaTime);
+            GameObject _preview = GameObject.FindWithTag("Release");
```
Comment (NGrinstead0101) [inline Assets/Scripts/Minigames/CowardMiniGame/RipcordBehavior.cs:75]: FindWithTag shouldn't be getting called repeatedly in Update. If possible, make this call once and store the result to use later.

File: Assets/Scripts/Minigames/CowardMiniGame/WrenchBehavior.cs [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+            transform.rotation = _rightHand.transform.rotation;
+        }
+        //if(_interact && _isEquipped == true)
+        //{
+        //    StartCoroutine(Swinging());
+        //}
+        //Debug.Log(_isEquipped);
+    }
+    // Update is called once per frame
+    void Update()
```
Comment (NGrinstead0101) [inline Assets/Scripts/Minigames/CowardMiniGame/WrenchBehavior.cs:75]: This Update function could be deleted to clean up the code a bit if it's not being used.

File: Assets/Scripts/Player/PlayerInteractSystem.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+// Author :            Mark Hanson
+// Creation Date :     5/22/2024
+//
+// Brief Description : Class used for detecting interactable objects 
+*****************************************************************************/
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerInteractSystem
```
Comment (NGrinstead0101) [inline Assets/Scripts/Player/PlayerInteractSystem.cs:12]: Is this class still needed? It seems to be redundant with Marissa's interaction system.

**Review by NGrinstead0101 — CHANGES_REQUESTED** [review]
See comments on scripts for more details.

File: Assets/Scripts/Player/PlayerInteractSystem.cs (hunk 6 above)
Comment (AlecatEnjoyer) [reply Assets/Scripts/Player/PlayerInteractSystem.cs:12]: Just old interaction system I'm sure this one isn't used anymore.

File: Assets/Scripts/Minigames/CowardMiniGame/WrenchBehavior.cs (hunk 5 above)
Comment (AlecatEnjoyer) [reply Assets/Scripts/Minigames/CowardMiniGame/WrenchBehavior.cs:75]: wrench script cleaned up

File: Assets/Scripts/Minigames/CowardMiniGame/RipcordBehavior.cs (hunk 3 above)
Comment (AlecatEnjoyer) [reply Assets/Scripts/Minigames/CowardMiniGame/RipcordBehavior.cs:26]: yeah I swap it to serialize field now

File: Assets/Scripts/Minigames/CowardMiniGame/GearCompletionCheck.cs (hunk 2 above)
Comment (AlecatEnjoyer) [reply Assets/Scripts/Minigames/CowardMiniGame/GearCompletionCheck.cs:6]: that was a blind site by me thank you for bringing that to my attention

File: Assets/Scripts/Minigames/CowardMiniGame/GearBehavior.cs (hunk 1 above)
Comment (AlecatEnjoyer) [reply Assets/Scripts/Minigames/CowardMiniGame/GearBehavior.cs:43]: added a bit more detail

**Pull request closed without merging**